src/serialization/interval_log/mod.rs

```diff
@@ -0,0 +1,587 @@
+//! epoch.
+//! - Use StartTime, and have each interval's timestamp be a delta from StartTime.
+//!
+//! Of course, if you are writing logs that need to work with an existing log processing pipeline,
+//! you should use timestamps as expected by that logic, so we provide the ability to have all
+//! combinations of timestamp headers if need be.
+//!
+//! # Examples
+//!
+//! Parse a single interval from a log.
+//!
+//! ```
+//! use hdrsample::serialization::interval_log;
+//!
+//! // two newline-separated log lines: a comment, then an interval
+//! let log = b"#I'm a comment\nTag=t,0.127,1.007,2.769,base64EncodedHisto\n";
+//!
+//! let mut iter = interval_log::IntervalLogIterator::new(&log[..]);
+//!
+//! // the comment is consumed and ignored by the parser, so the first event is an Interval
+//! match iter.next().unwrap().unwrap() {
+//!     interval_log::LogEntry::Interval(h) => {
+//!         assert_eq!(0.127, h.start_timestamp());
+//!     }
+//!     _ => panic!()
+//! }
+//!
+//! // there are no more lines in the log; iteration complete
+//! assert_eq!(None, iter.next());
+//! ```
+//!
+//! Skip logs that started before 3 seconds.
+//!
+//! ```
+//! use hdrsample::serialization::interval_log;
+//!
+//! let mut log = Vec::new();
+//! log.extend_from_slice(b"#I'm a comment\n");
+//! log.extend_from_slice(b"Tag=a,0.123,1.007,2.769,base64EncodedHisto\n");
+//! log.extend_from_slice(b"1.456,1.007,2.769,base64EncodedHisto\n");
+//! log.extend_from_slice(b"3.789,1.007,2.769,base64EncodedHisto\n");
+//! log.extend_from_slice(b"Tag=b,4.123,1.007,2.769,base64EncodedHisto\n");
+//! log.extend_from_slice(b"5.456,1.007,2.769,base64EncodedHisto\n");
+//! log.extend_from_slice(b"#Another comment\n");
+//!
+//! let iter = interval_log::IntervalLogIterator::new(&log);
+//!
+//! let count = iter.map(|r| r.unwrap())
+//!     // only look at intervals (which are the only non-comment lines in this log)
+//!     .filter_map(|e| match e {
+//!         interval_log::LogEntry::Interval(ilh) => Some(ilh),
+//!          _ => None
+//!     })
+//!     // do any filtering you want
+//!     .filter(|ilh| ilh.start_timestamp() >= 3.0)
+//!     .count();
+//!
+//! assert_eq!(3, count);
+//! ```
+//!
+//! Write a log.
+//!
+//! ```
+//! use std::str;
+//! use hdrsample;
+//! use hdrsample::serialization;
+//! use hdrsample::serialization::interval_log;
+//!
+//! let mut buf = Vec::new();
+//! let mut serializer = serialization::V2Serializer::new();
+//!
+//! let mut h = hdrsample::Histogram::<u64>::new_with_bounds(
+//!     1, u64::max_value(), 3).unwrap();
+//! h.record(12345).unwrap();
+//!
+//! // limit scope of mutable borrow of `buf`
+//! {
+//!     let mut header_writer = interval_log::IntervalLogHeaderWriter::new(
+//!         &mut buf, &mut serializer);
+//!     header_writer.write_comment("Comments are great").unwrap();
+//!     header_writer.write_start_time(123.456789).unwrap();
+//!
+//!     let mut log_writer = header_writer.into_log_writer();
```

Comment:
Haven't read further down yet, but is an `IntervalLogHeaderWriter` the only way to get an `IntervalLogWriter`? If so, I think the former should be given a different name.